### PR TITLE
FIXES #3558 - Merge profile_chooser_profile_view.xml into single xml file 

### LINKED ISSUE #3558: Merge profile_chooser_profile_view.xml into single xml file
Currently there are 4 versions of `profile_chooser_profile_view.xml` file merge it into single xml file.

We can use https://text-compare.com/ to compare two versions of this file and for all the differences we can create variables in `dimens.xml` file and use it accordingly.

It can happen that it becomes really hard to merge all 4 files into single file, in such cases prefer code-readability over single file. Therefore it can be fine if in the end the code still contains 2 or 3 files.

**Note**: **In PR, make sure you add before and after screenshot of mobile-portrait, mobile-landscape, tablet-portrait and tablet-landscape for comparison and make sure that there is not difference between before and after UI.**

### PR DESCRIPTION
**Pull request by Rishavgupta12345**
## Essential Checklist

- [x] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

## For UI-specific PRs only

If your PR includes UI-related changes, then:
- Add screenshots for portrait/landscape for both a tablet & phone of the before & after UI changes
- For the screenshots above, include both English and pseudo-localized (RTL) screenshots (see [RTL guide](https://github.com/oppia/oppia-android/wiki/RTL-Guidelines))
- Add a video showing the full UX flow with a screen reader enabled (see [accessibility guide](https://github.com/oppia/oppia-android/wiki/Accessibility-(A11y)-Guide))
- Add a screenshot demonstrating that you ran affected Espresso tests locally & that they're passing

fixes #3558

Before (Mobile-Portrait)
Screenshot 2021-11-25 121403
 
after (Mobile-Portrait)
Screenshot 2021-11-25 121422

Before (Mobile-landscape)
Screenshot 2021-11-25 121245

after (Mobile-landscape)
Screenshot 2021-11-25 121348

before (tab-landscape)
Screenshot 2021-11-25 121815

after (tab-landscape)
Screenshot 2021-11-25 121832

before (tab-potrait)
Screenshot 2021-11-25 121920

after (tab-potrait)
Screenshot 2021-11-25 121907

### CODE REVIEW

File: app/src/main/res/values-sw600dp-land/dimens.xml [hunk 1]
```diff
@@ -390,4 +390,6 @@
   <dimen name="solution_summary_margin_bottom_linear_layout">32dp</dimen>
   <dimen name="solution_summary_margin_top_solution_summary_label">16dp</dimen>
 
+
```
Comment (bkaur-bkj) [inline app/src/main/res/values-sw600dp-land/dimens.xml]: @Rishavgupta12345 remove these extra lines added, these are unnecessary file changes

File: app/src/main/res/layout/profile_chooser_profile_view.xml [hunk 2]
```diff
@@ -68,12 +69,12 @@
           android:id="@+id/profile_last_visited"
           android:layout_width="wrap_content"
           android:layout_height="wrap_content"
+          android:layout_marginTop="4dp"
           android:fontFamily="sans-serif-light"
+          android:gravity="@{hasProfileEverBeenAddedValue ? Gravity.CENTER_HORIZONTAL : Gravity.CENTER_VERTICAL}"
```
Comment (bkaur-bkj) [inline app/src/main/res/layout/profile_chooser_profile_view.xml]: @Rishavgupta12345 also in this file same statements have been just reordered, you can revert these changes if not needed

File: app/src/main/res/layout/profile_chooser_profile_view.xml [hunk 3]
```diff
@@ -19,6 +19,7 @@
       name="viewModel"
       type="org.oppia.android.app.model.ProfileChooserUiModel" />
   </data>
+
```
Comment (bkaur-bkj) [inline app/src/main/res/layout/profile_chooser_profile_view.xml]: @Rishavgupta12345 same here

**Comment by bkaur-bkj** [comment]
@rishav make sure after the changes are done you comment PTAL and tag reviewers so that reviewers will be assigned to review the issue

**Comment by bkaur-bkj** [comment]
@Rishavgupta12345 also pls update the title of thr PR with proper name it should be Fix #issue no and a statement explaining the fix or issue. Kindly take reference to other PRs which are merged

**Comment by Rishavgupta12345** [comment]
done the required changes now you can review PTAL @bkaur-bkj

**Comment by bkaur-bkj** [comment]
@Rishavgupta12345 make sure you answer and address all the review comments made by replying and don't resolve them yourself

File: app/src/main/res/values-sw600dp-land/dimens.xml [hunk 4]
```diff
@@ -389,5 +389,4 @@
   <dimen name="solution_summary_margin_top_linear_layout">0dp</dimen>
   <dimen name="solution_summary_margin_bottom_linear_layout">32dp</dimen>
   <dimen name="solution_summary_margin_top_solution_summary_label">16dp</dimen>
-
```
Comment (bkaur-bkj) [inline app/src/main/res/values-sw600dp-land/dimens.xml:392]: @Rishavgupta12345 this change is unnecessary here, as per the formatting there should be a blank line before</resources>

File: app/src/main/res/values-land/dimens.xml [hunk 5]
```diff
@@ -384,4 +384,10 @@
   <dimen name="profile_rename_activity_profile_rename_input_margin_end">168dp</dimen>
   <dimen name="profile_rename_activity_profile_rename_save_button_margin_end">168dp</dimen>
 
+  <!-- profile_chooser_profile_view -->
```
Comment (bkaur-bkj) [inline app/src/main/res/values-land/dimens.xml]: @Rishavgupta12345 here you don't need to use this snake case writing here, you should write it this way  this way like other headings are written.

File: app/src/main/res/values-sw600dp-port/dimens.xml [hunk 6]
```diff
@@ -392,4 +392,10 @@
   <dimen name="solution_summary_margin_bottom_linear_layout">32dp</dimen>
   <dimen name="solution_summary_margin_top_solution_summary_label">16dp</dimen>
 
+  <!-- profile_chooser_profile_view -->
```
Comment (bkaur-bkj) [inline app/src/main/res/values-sw600dp-port/dimens.xml]: @Rishavgupta12345 same here

File: app/src/main/res/values/dimens.xml [hunk 7]
```diff
@@ -584,4 +584,10 @@
   <dimen name="profile_rename_activity_profile_rename_input_margin_end">28dp</dimen>
   <dimen name="profile_rename_activity_profile_rename_save_button_margin_end">28dp</dimen>
 
+  <!-- profile_chooser_profile_view -->
```
Comment (bkaur-bkj) [inline app/src/main/res/values/dimens.xml]: @Rishavgupta12345 here too

**Review by bkaur-bkj — CHANGES_REQUESTED** [review]
@Rishavgupta12345 PTAL

**Comment by Rishavgupta12345** [comment]
done the required changes PTAL @bkaur-bkj

**Comment by bkaur-bkj** [comment]
> ## Explanation
> ## Essential Checklist
> * [ ]  The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
> * [ ]  Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
> * [ ]  The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
> * [ ]  The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
> * [ ]  The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
> * [ ]  The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).
> 
> ## For UI-specific PRs only
> If your PR includes UI-related changes, then:
> 
> * Add screenshots for portrait/landscape for both a tablet & phone of the before & after UI changes
> * For the screenshots above, include both English and pseudo-localized (RTL) screenshots (see [RTL guide](https://github.com/oppia/oppia-android/wiki/RTL-Guidelines))
> * Add a video showing the full UX flow with a screen reader enabled (see [accessibility guide](https://github.com/oppia/oppia-android/wiki/Accessibility-(A11y)-Guide))
> * Add a screenshot demonstrating that you ran affected Espresso tests locally & that they're passing
> 
> fixes #3558
> 
> Before (Mobile-Portrait) Screenshot 2021-11-25 121403
> 
> after (Mobile-Portrait) Screenshot 2021-11-25 121422
> 
> Before (Mobile-landscape) Screenshot 2021-11-25 121245
> 
> after (Mobile-landscape) Screenshot 2021-11-25 121348
> 
> before (tab-landscape) Screenshot 2021-11-25 121815
> 
> after (tab-landscape) Screenshot 2021-11-25 121832
> 
> before (tab-potrait) Screenshot 2021-11-25 121920
> 
> after (tab-potrait) Screenshot 2021-11-25 121907

@Rishavgupta12345 mark this checklist first as instructed [x] , put a 'x' in each [] after ensuring you have addressed each point

File: app/src/main/res/values-land/dimens.xml [hunk 8]
```diff
@@ -384,4 +384,10 @@
   <dimen name="profile_rename_activity_profile_rename_input_margin_end">168dp</dimen>
   <dimen name="profile_rename_activity_profile_rename_save_button_margin_end">168dp</dimen>
 
+  <!-- Profile Chooser Profile View -->
+  <dimen name="profile_chooser_profile_divider_view_layout_margin_top">28dp</dimen>
```
Comment (bkaur-bkj) [inline app/src/main/res/values-land/dimens.xml]: @Rishavgupta12345 all these dimens in these 3 files have not been used anywhere, just take a look at the PR and changes you have made , are they actually what is required?

Comment (bkaur-bkj) [reply app/src/main/res/values-land/dimens.xml]: use text compare to analyse the difference in portrait and landscape of the file and then make the required changes to merge and delete unused files, I did not get how have you merged the files as there was the difference in gravity and other parameters also

**Review by bkaur-bkj — CHANGES_REQUESTED** [review]

**Comment by Rishavgupta12345** [comment]
I have changed many things and done the things as per asked PTAL @bkaur-bkj  , please check and tell me if anything more needed further

**Comment by bkaur-bkj** [comment]
@Rishavgupta12345 this PR has some merge conflicts please resolve them by updating your branch with dev

File: app/src/main/res/layout/profile_chooser_profile_view.xml [hunk 9]
```diff
@@ -81,7 +81,7 @@
           android:id="@+id/profile_is_admin_text"
           android:layout_width="wrap_content"
           android:layout_height="wrap_content"
-          android:layout_marginTop="@dimen/profile_chooser_profile_view_is_admin_margin_top"
+          android:layout_marginTop="@dimen/profile_chooser_profile_text_view_layout_margin_top"
```
Comment (bkaur-bkj) [inline app/src/main/res/layout/profile_chooser_profile_view.xml:84]: @rt4914 sir can you pls review this part as I m having some doubt regarding this way of change

Comment (rt4914) [reply app/src/main/res/layout/profile_chooser_profile_view.xml:84]: @Rishavgupta12345 Please keep this previous name as it provides context about when to use to this margin
`profile_chooser_profile_view_is_admin_margin_top`

**Review by rt4914 — COMMENTED** [review]
@Rishavgupta12345 @bkaur-bkj Replied to Bhani's comment.

**Comment by rt4914** [comment]
@Rishavgupta12345 Please mention `Fixes #3558: Merge profile_chooser_profile_view files` just below `Explanation` and above `Essential Checklist`. Also the screenshots provided are incorrect. You should create screenshots for fresh copy of app when there is only admin (naming it as CASE A) and then with 3/4 profiles added (CASE B)
CASE A & CASE B - Before and After screenshots for mobile-portrait, mobile-landscape, tablet-portrait and tablet-landscape.
